perceval/backends/core/git.py

```diff
@@ -764,19 +763,26 @@ def _handle_trailer(self, line):
     def _handle_action_data(self, data):
         modes = self.__parse_data_list(data['modes'])
         indexes = self.__parse_data_list(data['indexes'])
-        filename = data['file']
+        filename = data['newfile'] if data['newfile'] else data['file']
 
         if filename not in self.commit_files:
             self.commit_files[filename] = {}
 
         self.commit_files[filename]['modes'] = modes
         self.commit_files[filename]['indexes'] = indexes
         self.commit_files[filename]['action'] = data['action']
-        self.commit_files[filename]['file'] = filename
+        self.commit_files[filename]['file'] = data['file']
         self.commit_files[filename]['newfile'] = data['newfile']
 
     def _handle_stats_data(self, data):
-        filename = self.__get_old_filepath(data['file'])
+        if ' => ' in data['file']:
+            filename = self.__get_new_filepath(data['file'])
+            if not self.merge_commit and self.commit_files and filename not in self.commit_files:
```

Comment:
Because this is a long check and hard to understand, maybe it's better to create a function that returns the result of this check:

```
def __should_guess_stats_filename(self):
    return not self.merge_commit and self.commit_files and filename not in self.commit_files
```